### ISSUE TITLE
unable to Retrieve data in android app from Dynamodb

### ISSUE THREAD

**Issue by JagritiHirwani** [awslabs/aws-mobile-appsync-sdk-android#120]
**how to get data from amazon dynamodb table into the mobile app. I have followed aws mobile hub and amplify documentations but nothing works.**
My mainactivity.java looks like this:

```
package com.example.lenovo.myfarm;


import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;
import android.widget.EditText;
import android.widget.TextView;

import com.amazonaws.amplify.generated.graphql.CreateTodoMutation;
import com.amazonaws.mobile.client.AWSMobileClient;
import com.amazonaws.mobile.config.AWSConfiguration;

import com.amazonaws.mobile.config.AWSConfiguration;
import com.amazonaws.mobileconnectors.appsync.AWSAppSyncClient;
import com.amazonaws.mobileconnectors.appsync.AppSyncSubscriptionCall;
import com.amazonaws.mobileconnectors.dynamodbv2.document.Expression;
import com.amazonaws.mobileconnectors.dynamodbv2.document.Search;
import com.amazonaws.services.dynamodbv2.model.AttributeValue;
import com.amazonaws.services.dynamodbv2.model.GetItemRequest;
import com.amazonaws.services.dynamodbv2.model.GetItemResult;
import com.apollographql.apollo.GraphQLCall;
import com.apollographql.apollo.api.Response;
import com.apollographql.apollo.exception.ApolloException;
import com.amazonaws.mobileconnectors.dynamodbv2.document.Table;
import com.amazonaws.mobileconnectors.dynamodbv2.document.UpdateItemOperationConfig;
import com.amazonaws.mobileconnectors.dynamodbv2.document.datatype.Document;
import com.amazonaws.mobileconnectors.dynamodbv2.document.datatype.Primitive;
import com.amazonaws.services.dynamodbv2.AmazonDynamoDBClient;
import com.amazonaws.services.dynamodbv2.model.ReturnValue;

import javax.annotation.Nonnull;
import com.amazonaws.auth.CognitoCachingCredentialsProvider;
import com.amazonaws.services.dynamodbv2.AmazonDynamoDBClient;
import com.amazonaws.services.dynamodbv2.model.ReturnValue;

import type.CreateTodoInput;
import com.amazonaws.mobileconnectors.dynamodbv2.dynamodbmapper.DynamoDBMapper;
import com.amazonaws.services.dynamodbv2.AmazonDynamoDBClient;

import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class MainActivity extends AppCompatActivity {
    private AWSAppSyncClient mAWSAppSyncClient;
    private AppSyncSubscriptionCall subscriptionWatcher;
    DynamoDBMapper dynamoDBMapper;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        mAWSAppSyncClient = AWSAppSyncClient.builder()
                .context(getApplicationContext())
                .awsConfiguration(new AWSConfiguration(getApplicationContext()))
                .build();
        AmazonDynamoDBClient dynamoDBClient = new AmazonDynamoDBClient(AWSMobileClient.getInstance().getCredentialsProvider());
        this.dynamoDBMapper = DynamoDBMapper.builder()
                .dynamoDBClient(dynamoDBClient)
                .awsConfiguration(
                        AWSMobileClient.getInstance().getConfiguration())
                .build();
        TextView textView = (TextView) findViewById(R.id.textView);
        Table myTable = Table.loadTable(dynamoDBClient, "Todo-3vblnrl4zvgorcef72bfxkkoka");
        Document retrievedDoc = myTable.getItem(new Primitive("id"));
        textView.setText(retrievedDoc.get(“temperature”).asString());
    }
}
```

[https://aws.amazon.com/blogs/mobile/using-amazon-dynamodb-document-api-with-aws-mobile-sdk-for-android-part-1/](I followed this, but was not successful)

my dynamodb table consists of id (string) as hash key and two coloumns temperature and humidity (float). I want to retrieve this data and show in my app.

**Environment(please complete the following information):**
 - AppSync SDK Version: 2.7
**Device Information (please complete the following information):**
 - Device: [e.g. Pixel XL, Simulator]
 - Android Version: 9
 - Specific to simulators:

**Comment by kvasukib** [comment]
@JagritiHirwani Sorry that you are running into issues. Can you explain the issue you are facing? Are you able to send data to your DynamoDB Table?

**Comment by cbommas** [comment]
@JagritiHirwani 
Just curious about why you are accessing DynamoDB directly instead of using AppSync.

For the issue at hand, can you share logs and stack traces?

**Comment by JagritiHirwani** [comment]
@cbommas 
Sorry for replying so late.

I have created an AppSync Api in my aws and imported a DynamoDB table containing humidity and temperature values.  I followed the steps given [here](https://aws-amplify.github.io/docs/android/api).

The code in MainAcitivity is as follows:

`public void query(){
        mAWSAppSyncClient.query(ListTemperatureHumidityTablesQuery.builder().build())
                .responseFetcher(AppSyncResponseFetchers.NETWORK_FIRST)
                .enqueue(todosCallback);
    }
    private GraphQLCall.Callback<ListTemperatureHumidityTablesQuery.Data> todosCallback = new GraphQLCall.Callback<ListTemperatureHumidityTablesQuery.Data>() {
        @Override
        public void onResponse(@Nonnull Response<ListTemperatureHumidityTablesQuery.Data> response) {
            Log.i("Results", response.data().listTemperatureHumidityTables().items().toString());
        }

        @Override
        public void onFailure(@Nonnull ApolloException e) {
            Log.e("ERROR", e.toString());
        }
    };`

My logcat shows the following error:
`2019-03-09 16:54:49.359 1790-1812/system_process E/memtrack: Couldn't load memtrack module
2019-03-09 16:54:49.359 1790-1812/system_process W/android.os.Debug: failed to get memory consumption info: -1
2019-03-09 16:54:54.360 2831-2845/com.google.android.gms.persistent I/zygote: Background concurrent copying GC freed 147401(6MB) AllocSpace objects, 0(0B) LOS objects, 38% free, 9MB/15MB, paused 14.272ms total 287.507ms
2019-03-09 16:54:56.867 6532-6550/com.example.lenovo.app2 V/AppSyncOfflineMutationInterceptor: Thread:[325]: processing Mutations
2019-03-09 16:54:56.868 6532-6550/com.example.lenovo.app2 V/QueueUpdateHandler: Thread:[325]: Got message to take action on the mutation queue.
2019-03-09 16:54:56.869 6532-6550/com.example.lenovo.app2 V/QueueUpdateHandler: Thread:[325]: Got message to process next mutation if one exists.
2019-03-09 16:54:56.871 6532-6550/com.example.lenovo.app2 V/AppSyncOfflineMutationManager: Thread:[325]:Persistent mutations queue is EMPTY!. Will check inMemory Queue next
2019-03-09 16:54:56.871 6532-6550/com.example.lenovo.app2 V/AppSyncOfflineMutationManager: Thread:[325]: In Memory mutations queue was EMPTY!. Nothing to process, exiting
`
I am unable to query the data as of now and i do not get any result.
